src/mlmc/quantity.py

```diff
@@ -967,12 +1079,15 @@ def base_qtype(self):
     def size(self) -> int:
         return int(np.sum(q_type.size() for _, q_type in self._dict.items()))
 
-    def replace_scalar(self, new_qtype):
-        for key, qtype in self._dict.items():
+    @staticmethod
```

Comment:
How is this related to QType.replace_scalar? It is not even a classmethod. It seems to be already in use, but introduction of the probably same static method in QType is confusing.
Should be probably merged to a single method probably with virtual implementation in some QType subclases.